compiler/src/iree/compiler/Dialect/Flow/Transforms/Passes.h

```diff
@@ -176,6 +176,10 @@ createOutlineDispatchRegionsPass();
 std::unique_ptr<InterfacePass<mlir::FunctionOpInterface>>
 createInjectDispatchTracingPass();
 
+// Crops the dispatch calls at the given index (-1 for nothing)
+std::unique_ptr<InterfacePass<mlir::FunctionOpInterface>>
+createCropDispatchPipelinePass(int dispatchIndex = -1);
```

Comment:
There was an RFC posted this week, and discussed at the MLIR open meeting yesterday, that may offer a more general way to implement this sort of compiler debugging: https://discourse.llvm.org/t/rfc-mlir-action-tracing-and-debugging-mlir-based-compilers/68679 ([slides/recording linked here](https://discourse.llvm.org/t/open-mlir-meeting-2-23-2023-mlir-actions-tracing-and-debugging-mlir-based-compilers/68680)). With that you could filter actions (pattern applications, passes, etc.) and choose to skip some - possibly even without needing to write C++ code or recompile the compiler.

Reply:
Oh cool, having these tools would probably eliminate the use case for this pass to debug compiler failures. Although I think having the ability to jump in and look at the output of a specific dispatch would be hard to do with this. Perhaps that's closer to something we want to be able to do in the runtime? So that the model doesn't need to be recompiled just to look at the output of a neighboring dispatch.